### ISSUE TITLE
Bridge exited signal: abort trap

### ISSUE THREAD

**Issue by dsaderholm** [ngencokamin/sh-imessage-setup#13]
I'm getting this error after the script attempts to run the bridge. I am running a pretty old version of iMessage, so that could be the cause. Any help would be appreciated!

```
Successfully installed mautrix-imessage commit e37cea98
Starting imessage
dyld: Library not loaded: @rpath/libolm.3.dylib
Referenced from: /Users/ME/Library/Application Support/bbctl/prod/binaries/mautrix-imessage
Reason: no suitable image found. Did find:
/Users/ME/Library/Application Support/bbctl/prod/binaries/libolm.3.dylib: cannot load 'libolm.3.dylib' (load command 0x80000034 is unknown)
/Users/ME/Library/Application Support/bbctl/prod/binaries/libolm.3.dylib: stat() failed with errno=1
Bridge exited
signal: abort trap
```

**Comment by MattDemers** [comment]
Just started having this problem today, as well.

**Comment by ngencokamin** [comment]
@dsaderholm @MattDemers ok I will look into this. technically that'd be a bridge issue, not a script issue. I'll see if there's anything I can figure out tho. 

Can you guys pls share the details of the systems you're on. general model, archicture (whether it's m series or not), and macos version? I havent reproduced this on my 2012 Mac mini running ventura, so will have to spin up a docker container and see if I can reproduce if you're on the newest version or smth.

**Comment by MattDemers** [comment]
I'm using a mid-2012 Macbook Pro on Catalina (10.15.7).
```
CPU: 2.4 GHz Quad-Core Intel Core i7
Memory: 8gb 1600 MHz DDR3
Graphics: Nvidia GeForce GT 650M 512 MB
```
^ Probably overkill, but who knows.

When I try to install libolm with `brew install libolm` I get:

```
You have 1 outdated formula installed.

Warning: You are using macOS 10.15.
We (and Apple) do not provide support for this old version.

This is a Tier 3 configuration:
  https://docs.brew.sh/Support-Tiers#tier-3
You can report Tier 3 unrelated issues to Homebrew/* repositories!
Read the above document instead before opening any issues or PRs.

==> Fetching downloads for: libolm
Warning: libolm has been deprecated because it is deprecated upstream! It was disabled on 2025-08-03.
==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/07b1aade4a032d15821ac354241572d25e72
###################################################################################################### 100.0%
==> Fetching libolm
==> Downloading https://gitlab.matrix.org/matrix-org/olm/-/archive/3.2.16/olm-3.2.16.tar.gz
Already downloaded: /Users/matt/Library/Caches/Homebrew/downloads/2cbaff58e14945c51a9e15c9835b9a6b9031f6eb4a210da6555c1840d8e9c10d--olm-3.2.16.tar.gz
==> cmake -S . -B build
Last 15 lines from /Users/matt/Library/Logs/Homebrew/libolm/01.cmake:
-Wno-dev
-DBUILD_TESTING=OFF
-DCMAKE_OSX_SYSROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk

CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.


-- Configuring incomplete, errors occurred!



Error: You are using macOS 10.15.
We (and Apple) do not provide support for this old version.

This is a Tier 3 configuration:
  https://docs.brew.sh/Support-Tiers#tier-3
You can report Tier 3 unrelated issues to Homebrew/* repositories!
Read the above document instead before opening any issues or PRs.

This build failure was expected, as this is not a Tier 1 configuration:
  https://docs.brew.sh/Support-Tiers
Do not report any issues to Homebrew/* repositories!
Read the above document instead before opening any issues or PRs.
```

I think this is the closest thing to information I have that may be helpful.

Not sure if this is fixable - unfortunately I've been trying to update my Mac with OpenCore Legacy Patcher and running into separate issues there, in order to see if the problem is just my MacOS version being too old 😒.

**Comment by ngencokamin** [comment]
> I'm using a mid-2012 Macbook Pro on Catalina (10.15.7).
> ```
> CPU: 2.4 GHz Quad-Core Intel Core i7
> Memory: 8gb 1600 MHz DDR3
> Graphics: Nvidia GeForce GT 650M 512 MB
> ```
> ^ Probably overkill, but who knows.
> 
> When I try to install libolm with `brew install libolm` I get:
> 
> ```
> You have 1 outdated formula installed.
> 
> Warning: You are using macOS 10.15.
> We (and Apple) do not provide support for this old version.
> 
> This is a Tier 3 configuration:
>   https://docs.brew.sh/Support-Tiers#tier-3
> You can report Tier 3 unrelated issues to Homebrew/* repositories!
> Read the above document instead before opening any issues or PRs.
> 
> ==> Fetching downloads for: libolm
> Warning: libolm has been deprecated because it is deprecated upstream! It was disabled on 2025-08-03.
> ==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/07b1aade4a032d15821ac354241572d25e72
> ###################################################################################################### 100.0%
> ==> Fetching libolm
> ==> Downloading https://gitlab.matrix.org/matrix-org/olm/-/archive/3.2.16/olm-3.2.16.tar.gz
> Already downloaded: /Users/matt/Library/Caches/Homebrew/downloads/2cbaff58e14945c51a9e15c9835b9a6b9031f6eb4a210da6555c1840d8e9c10d--olm-3.2.16.tar.gz
> ==> cmake -S . -B build
> Last 15 lines from /Users/matt/Library/Logs/Homebrew/libolm/01.cmake:
> -Wno-dev
> -DBUILD_TESTING=OFF
> -DCMAKE_OSX_SYSROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk
> 
> CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
>   Compatibility with CMake < 3.5 has been removed from CMake.
> 
>   Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
>   to tell CMake that the project requires at least <min> but has been updated
>   to work with policies introduced by <max> or earlier.
> 
>   Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
> 
> 
> -- Configuring incomplete, errors occurred!
> 
> 
> 
> Error: You are using macOS 10.15.
> We (and Apple) do not provide support for this old version.
> 
> This is a Tier 3 configuration:
>   https://docs.brew.sh/Support-Tiers#tier-3
> You can report Tier 3 unrelated issues to Homebrew/* repositories!
> Read the above document instead before opening any issues or PRs.
> 
> This build failure was expected, as this is not a Tier 1 configuration:
>   https://docs.brew.sh/Support-Tiers
> Do not report any issues to Homebrew/* repositories!
> Read the above document instead before opening any issues or PRs.
> ```
> 
> I think this is the closest thing to information I have that may be helpful.
> 
> Not sure if this is fixable - unfortunately I've been trying to update my Mac with OpenCore Legacy Patcher and running into separate issues there, in order to see if the problem is just my MacOS version being too old 😒.

My best guess is macos is too behind. i *think* it's labeling you as tier 3 because the package is deprecated. Lemme see what happens when I do this on my ventura Mac rq.

**Comment by ngencokamin** [comment]
> I'm using a mid-2012 Macbook Pro on Catalina (10.15.7).
> ```
> CPU: 2.4 GHz Quad-Core Intel Core i7
> Memory: 8gb 1600 MHz DDR3
> Graphics: Nvidia GeForce GT 650M 512 MB
> ```
> ^ Probably overkill, but who knows.
> 
> When I try to install libolm with `brew install libolm` I get:
> 
> ```
> You have 1 outdated formula installed.
> 
> Warning: You are using macOS 10.15.
> We (and Apple) do not provide support for this old version.
> 
> This is a Tier 3 configuration:
>   https://docs.brew.sh/Support-Tiers#tier-3
> You can report Tier 3 unrelated issues to Homebrew/* repositories!
> Read the above document instead before opening any issues or PRs.
> 
> ==> Fetching downloads for: libolm
> Warning: libolm has been deprecated because it is deprecated upstream! It was disabled on 2025-08-03.
> ==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/07b1aade4a032d15821ac354241572d25e72
> ###################################################################################################### 100.0%
> ==> Fetching libolm
> ==> Downloading https://gitlab.matrix.org/matrix-org/olm/-/archive/3.2.16/olm-3.2.16.tar.gz
> Already downloaded: /Users/matt/Library/Caches/Homebrew/downloads/2cbaff58e14945c51a9e15c9835b9a6b9031f6eb4a210da6555c1840d8e9c10d--olm-3.2.16.tar.gz
> ==> cmake -S . -B build
> Last 15 lines from /Users/matt/Library/Logs/Homebrew/libolm/01.cmake:
> -Wno-dev
> -DBUILD_TESTING=OFF
> -DCMAKE_OSX_SYSROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk
> 
> CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
>   Compatibility with CMake < 3.5 has been removed from CMake.
> 
>   Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
>   to tell CMake that the project requires at least <min> but has been updated
>   to work with policies introduced by <max> or earlier.
> 
>   Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
> 
> 
> -- Configuring incomplete, errors occurred!
> 
> 
> 
> Error: You are using macOS 10.15.
> We (and Apple) do not provide support for this old version.
> 
> This is a Tier 3 configuration:
>   https://docs.brew.sh/Support-Tiers#tier-3
> You can report Tier 3 unrelated issues to Homebrew/* repositories!
> Read the above document instead before opening any issues or PRs.
> 
> This build failure was expected, as this is not a Tier 1 configuration:
>   https://docs.brew.sh/Support-Tiers
> Do not report any issues to Homebrew/* repositories!
> Read the above document instead before opening any issues or PRs.
> ```
> 
> I think this is the closest thing to information I have that may be helpful.
> 
> Not sure if this is fixable - unfortunately I've been trying to update my Mac with OpenCore Legacy Patcher and running into separate issues there, in order to see if the problem is just my MacOS version being too old 😒.

Okay so it worked on ventura, and did not give any messages Abt my macos being too old. What's going wrong with OCLP? Maybe I can help debug?

Btw, you're gonna wanna go to Ventura, not higher. Besides hardware requirements and decreasing Intel support, it's the least broken with the new emoji reactions.

**Comment by MattDemers** [comment]
**Update**: I've gotten farther than I ever have before after doing the following:

1. Restarting, leaving the USB in
2. Opening Disk Utility and trying to "First Aid" the disk. This gave me a similar (but different) error: 69673. This is "Unable to unmount volume for repair" 
3. I did the method of Option-Clicking Finder, and hitting "Relaunch"
4. I tried to repair again, and it worked
5. I then tried to use the installer again - It's at 265.88MB, which is farther than it's ever gone.

---

> Okay so it worked on ventura, and did not give any messages Abt my macos being too old. What's going wrong with OCLP? Maybe I can help debug?
> 
> Btw, you're gonna wanna go to Ventura, not higher. Besides hardware requirements and decreasing Intel support, it's the least broken with the new emoji reactions.

Forgive my over-describing, but I'm kinda new to all this.

I'm using OpenCore Legacy Patcher; I have it on my Macbook, and I'm looking at it right now.

- I'm clicking "Create macOS Installer", and selecting Ventura. I downloaded a fresh copy, it validates all the chunks, etc.
- I have a USB key inserted into the one working USB slot in the laptop. It's a 32gb size, which I know is like, the minimum? It shows up as 31gb in the patcher.
- When I try to create the bootable media, it warns me about deleting everything on the disk (which is fine).
- I confirm this, and it starts; the MB/s is pretty low (either fractions of an MB or 2-3, every 10ish seconds).
- Eventually, around 100-150MB, I get an error, which ends the install media creation: 

<img width="410" height="362" alt="Image" src="https://github.com/user-attachments/assets/9cf261b0-7eb6-4eea-8ca1-837a650a641e" />

Basically from what I can tell, multiple programs are trying to use my USB, so it's interfering here. I've tried running `diskutil eject disk2`, which ejects it but OpenCore can't see it then (duh, I guess). Have also tried running `sudo fuser -c /dev/disk2` (from [these instructions](https://apple.stackexchange.com/questions/389052/error-69888-couldnt-unmount-disk)). It returns nothing in terminal.

**Comment by dsaderholm** [comment]
I just wanted to let you guys know that I have switch to OpenBubbles primarily, but I do have BlueBubbles running on that Mac still as a backup. Let me know if you need anything from me. I haven't really used that Mac in a while now. It is a Mac mini server from mid 2010. Yeah, it is that old.

**Comment by MattDemers** [comment]
Just wanted to close the loop on this for me:

Updating macOS and running `brew install libolm` worked for me.
